pytest_run.py

```diff
@@ -11,7 +11,7 @@
 
 https://sopel.chat
 """
-from __future__ import absolute_import, division, print_function, unicode_literals
+from __future__ import generator_stop
```

Comment:
I am not going to mark every one of these, but what's the chance of subtle behavior changes from adding this? (Yes, I realize it's "mandatory" on three of our four currently-supported Python versions, 3.7+.)

Reply:
I didn't know before looking into it, so here is the result of my research:

> The interaction of generators and StopIteration is currently somewhat surprising, and can conceal obscure bugs. [...] Currently, StopIteration raised accidentally inside a generator function will be interpreted as the end of the iteration by the loop construct driving the generator.

This quote is from the [PEP 479](https://www.python.org/dev/peps/pep-0479/) (you can read it for more details). The idea is that the `__future__` import prevents to stop a loop with a `StopIteration` exception within a generator: it'll be raised as a normal exception and handled up in the stack.